docs/ingress-gateway-asm-manifests/base/deployment-service.yaml

```diff
@@ -0,0 +1,130 @@
+kind: ServiceAccount
+metadata:
+  name: asm-ingressgateway
+# [END servicemesh_base_deployment_service_serviceaccount_asm_ingressgateway]
+---
+# [START servicemesh_base_deployment_service_deployment_asm_ingressgateway]
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: asm-ingressgateway
+spec:
+  selector:
+    matchLabels:
+      asm: ingressgateway
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels:
+        asm: ingressgateway
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto
+        env:
+        - name: ISTIO_META_UNPRIVILEGED_POD
+          value: "true"
+        ports:
+        - containerPort: 8080
+          protocol: TCP
+        resources:
+          limits:
+            cpu: 2000m
+            memory: 1024Mi
+          requests:
+            cpu: 100m
+            memory: 128Mi
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+              - all
+          privileged: false
+          readOnlyRootFilesystem: true
+      securityContext:
+        fsGroup: 1337
+        runAsGroup: 1337
+        runAsNonRoot: true
+        runAsUser: 1337
+      serviceAccountName: asm-ingressgateway
+# [END servicemesh_base_deployment_service_deployment_asm_ingressgateway]
+---
+# [START servicemesh_base_deployment_service_service_asm_ingressgateway]
+apiVersion: v1
+kind: Service
+metadata:
+  name: asm-ingressgateway
+  labels:
+    asm: ingressgateway
+spec:
+  ports:
+  - name: http
+    port: 80
+    targetPort: 8080
+  selector:
+    asm: ingressgateway
+  type: LoadBalancer
+# [END servicemesh_base_deployment_service_service_asm_ingressgateway]
+---
+# [START servicemesh_base_deployment_service_hpa_asm_ingressgateway]
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: asm-ingressgateway
+spec:
+  maxReplicas: 5
+  minReplicas: 3
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      target:
+        type: Utilization
+        averageUtilization: 50
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: asm-ingressgateway
+# [END servicemesh_base_deployment_service_hpa_asm_ingressgateway]
+---
+# [START servicemesh_base_deployment_service_role_asm_ingressgateway]
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: asm-ingressgateway
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "watch", "list"]
+# [END servicemesh_base_deployment_service_role_asm_ingressgateway]
+---
+# [START servicemesh_base_deployment_service_rolebinding_asm_ingressgateway]
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: asm-ingressgateway
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: asm-ingressgateway
+subjects:
+  - kind: ServiceAccount
+    name: asm-ingressgateway
+# [END servicemesh_base_deployment_service_rolebinding_asm_ingressgateway]
```

Comment:
**Thought:**
I wonder if we should do the same for https://github.com/GoogleCloudPlatform/anthos-service-mesh-samples/tree/main/docs/shared/asm-ingress-gateway?
No action required (inside this PR).

Reply:
Yep good point, we let @xtineskim decide, the context of that is that as best practice we should show/have this. We discussed that with a CEs based on their customer feedback on that guide: https://cloud.google.com/architecture/exposing-service-mesh-apps-through-gke-ingress, so took the opportunity to address that here too.